pkg/pbq/interfaces.go

```diff
@@ -0,0 +1,28 @@
+package pbq
+
+import (
+	"context"
+	"github.com/numaproj/numaflow/pkg/isb"
+)
+
+// Reader provides methods to read from PBQ.
+type Reader interface {
```

Comment:
Based on my understanding, the the PBQ is a channel concept, the read and write happen in same application, and read and write even share the same instance. Then it seems to make no sense to have 2 interfaces defined.

Also, based on the code below, there's only one implementation, it's not related to any type of `store` (store is an interface), and your `NewPBQ` returns an `*PBQ` instead of an interface, then what's the point of having interfaces defined here?

Reply:
Maybe merging these 2 interfaces makes more sense?

Reply:
yes, i think we should merge both `Reader` and `Writer` interface into `ReaderWriter` interface. It will look more cleaner. We really do not have to worry about the separation of concern because we are not leaking this to SDK.

Reply:
As of now, its only one implementation, but depending on how  windowing shapes up, we may have to have composite readers vs single readers etc. 

one of the reasons why reader and writer are two interfaces is to make sure that we give access to behavior that is expected instead of throwing open all the functions. also the intent becomes clear.

can we merge these two later? if needed, we can introduce a new interface which composes both reader and writer something like `PBQReadWriter`

Reply:
To the comment about *PBQ, i agree. returning a pointer when you have interfaces defined probably doesn't make a lot of sense. PBQManager could instead expose methods for reader/writer

Reply:
@ashwinidulams what do you mean by composite readers?

Reply:
@yhl25 some refactoring has been made.. please merge https://github.com/yhl25/numaflow/pull/1

Reply:
the interface now allows you to do 
```golang
pbq := NewPBQ() ReaderWriter
// for left side
ReadLoop(pbq.(PBQWriter))
// for right loop
ProcessAndForward(pbq.(PBQReader))
```

Reply:
in case of session windowing, depending on the order in which elements arrive, one could have isolated windows to begin with which may need to be merged (once the elements in the gaps arrive). To handle such a situation its not easy to move the data around and so a partitioner will be required to orchestrate elements from multiple partitions to be read by a single reader.  a Pbq reader will be required to support reading from multiple partitions thereby creating a fan-in scenario. this is what i mean by a composite reader.